android/app/src/main/java/com/chorestory/app/ParentLoginActivity.java

```diff
@@ -40,18 +61,35 @@ public void onClick(View v) {
                 username = usernameEditText.getText().toString();
                 password = passwordEditText.getText().toString();
 
-                // TODO: Authenticate user
+                LoginRequest loginRequest = new LoginRequest(username, password);
+                Call<LoginResponse> loginQuery = retrofitInterface.login(loginRequest);
+
+                loginQuery.enqueue(new Callback<LoginResponse>() {
+                    @Override
+                    public void onResponse(Call<LoginResponse> call, Response<LoginResponse> response) {
+                        if (response.isSuccessful()) {
+                            String token = response.body().getToken();
 
-                // TODO: if authentication successful, navigate to app homepage
+                            // TODO: Store the token properly here
+                            tokenHandler.setToken(token);
 
-                // TODO: if authentication unsuccessful, enable loginButton, display toast
-                // TODO: use Snackbar instead; move existing view up when Snackbar appears
-//                toast = Toast.makeText(ParentLoginActivity.this,
-//                        "Invalid username or password",
-//                        Toast.LENGTH_SHORT);
-//                toast.setGravity(Gravity.CENTER_HORIZONTAL, 0, 400);
-//                toast.show();
-//                logInButton.setEnabled(true);
+                            // TODO: Navigate to profile page
+                        } else{
```

Comment:
in `Android Studio` 
hover over `Code` menu
click `Reformat Code` 
on all modified files please